### ISSUE TITLE
linter: `no-control-regex` false positive

### ISSUE THREAD

**Issue by pumano** [oxc-project/oxc#6525]
### What version of Oxlint are you using?

0.9.10

### What command did you run?

oxlint -c oxlinttest.json --quiet

### What does your `.oxlint.json` config file look like?

```json

{
  "files": [".ts"],
    "rules": {
      "no-control-regex": "error"
  }
}
```

### What happened?
```js
const filenameRegex = /filename[^;=\n]=((['"]).?\2|[^;\n]*)/;
```

```
  × eslint(no-control-regex): Unexpected control character(s)
    ╭─[features/drivers/src/lib/services/download.service.ts:22:29]
 21 │       // TODO: oxlint error here
 22 │       const filenameRegex = /filename[^;=\n]*=((['"]).*?\2|[^;\n]*)/;
    ·                             ────────────────────────────────────────
 23 │       const matches = filenameRegex.exec(content);
    ╰────
  help: Unexpected control character(s) in regular expression: "\2"
  ```
  
  eslint mark that code - OK
  I also test that regular expression against eslint playground: https://eslint.org/docs/latest/rules/no-control-regex - it's OK

**Comment by camc314** [comment]
fixed in #6529
